plugins/modules/source_control/github/github_repo.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Álvaro Torres Cogollo
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: github_repo
+short_description: Manage your repositories on Github
+version_added: 2.1.0
+description:
+- Manages Github repositories using PyGithub library.
+- Authentication can be done with I(access_token) or with I(username) and I(password).
+options:
+  username:
+    description:
+    - Username used for authentication.
+    - This is only needed when not using I(access_token).
+    type: str
+    required: false
+  password:
+    description:
+    - Password used for authentication.
+    - This is only needed when not using I(access_token).
+    type: str
+    required: false
+  access_token:
+    description:
+    - Token parameter for authentication.
+    - This is only needed when not using I(username) and I(password).
+    type: str
+    required: false
+  name:
+    description:
+    - Repository name.
+    type: str
+    required: true
+  description:
+    description:
+    - Description for the repository.
+    - This is only used when I(state) is C(present).
+    type: str
+    default: ''
+    required: false
+  private:
+    description:
+    - Whether the new repository should be private or not.
+    - This is only used when I(state) is C(present).
+    type: bool
+    default: no
+    required: false
+  state:
+    description:
+    - Whether the repository should exist or not.
+    type: str
+    default: present
+    choices: [ absent, present ]
+    required: false
+  organization:
+    description:
+    - Organization for the repository.
+    - When I(state) is C(present), the repository will be created in the current user profile.
+    type: str
+    required: false
+requirements:
+- PyGithub>=1.54
+notes:
+- For python3, PyGithub>=1.54 should be used.
+- "For python3.5, PyGithub==1.54 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-54-november-30-2020."
+- "For python2.7, PyGithub==1.45 should be used. More info: https://pygithub.readthedocs.io/en/latest/changes.html#version-1-45-december-29-2019."
+author:
+- Álvaro Torres Cogollo (@atorrescogollo)
+'''
+
+EXAMPLES = '''
+- name: Create a Github Repo
+  community.general.github_repo:
+    access_token: mytoken
+    organization: MyOrganization
+    name: myrepo
+    description: "Just for fun"
+    private: yes
+    state: present
+  register: result
+- name: Delete the repo
```

Comment:
The newline between the examples makes them a bit easier to read:
```suggestion

- name: Delete the repository
```